### PR TITLE
Clean more BuildFiles in some subsystems

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick partially automatic BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/31685).

This PR also moves some library BuildFiles from the `src` directory to the root of the package where they belong.

As always, only the dependencies which are **not included in any of the sources** are removed, so these changes are orthogonal and complementary to the recent PRs which added all packages that are **actually included**.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

File: CalibTracker/SiStripCommon/BuildFile.xml [hunk 1]
```diff
@@ -4,7 +4,6 @@
 <use name="DataFormats/SiStripDetId"/>
 <use name="DataFormats/Scalers"/>
 <use name="DataFormats/OnlineMetaData"/>
-<use name="SimDataFormats/TrackingAnalysis"/>
```
Comment (mmusich) [inline CalibTracker/SiStripCommon/BuildFile.xml:7]: @guitargeek that's gonna conflict with https://github.com/cms-sw/cmssw/pull/31643. Hopefully after staying idle for 15 days will be merged...

Comment (guitargeek) [reply CalibTracker/SiStripCommon/BuildFile.xml:7]: Thanks for letting me know! I'll get the CalibTracker changes out of this PR then.

**Comment by civanch** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by cvuosalo** [comment]
+1

**Comment by guitargeek** [comment]
@ggovi, are these changes okay for you?

**Comment by ggovi** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**